src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -316,9 +336,92 @@ private string FormatEnumValueToString(string value, JavaScriptEncoder? encoder)
             return converted;
         }
 
+        private static string[] SplitFlagsEnum(string value)
```

Comment:
Nit: private helpers should be placed be below converter overrides.

Reply:
I think this could also incorporate the logic that naming converts and re-joins the tokens?